### ISSUE TITLE
Should resolve relative URLs provided in Link headers.

### ISSUE THREAD

**Issue by Zegnat** [grantcodes/micropub#7]
I am sorry, but I have returned. Looks like we missed another relative URL problem. `Link` headers can be relative, just link the `href` values on `link` elements:

```
Link: </micropub>; rel="micropub"
```

This is currently not supported because only rel-scraper resolves URLs.

I hope to find time to fix this soonish, but no promises, so if everyone else can take it up: please!

**Comment by grantcodes** [comment]
Ok I think I have a fix for this published on the github, but I've not tested it (or published to npm)

So if you can test let me know.

**Comment by Zegnat** [comment]
From my testing (using the master branch here as a dependency in Omnibear), this fixes relative URLs in headers! 🎉 

Unrelated to relative URLs, but my testing turned up a small bug in the rel matching logic: https://github.com/grantcodes/micropub/blob/b725f4fa39d570b8f3a058f15ecdbf25f0f86f4b/src/main.js#L101-L102

This will happily accept a header like:

```
Link: </nothing>; rel="foo_micropub_bar"
```

You either need to split `rel[1]` on spaces (` `) and check the resulting array, or do something like:

```javascript
(' ' + rel[1] + ' ').indexOf(' ' + key + ' ') >= 0
```

Note that according to [RFC 8288 section 2.1.1](https://tools.ietf.org/html/rfc8288#section-2.1.1), relations “MUST be compared <i>[…]</i> in a case-insensitive fashion”. So probably best to drop in a `toLowerCase()` somewhere as well.

**Comment by grantcodes** [comment]
Ok should all be fixed in the 1.3.10 release.
